### PR TITLE
Issue 8

### LINKED ISSUE #8: Add branch picker to in-app updater
Currently, when the 'unstable' update option is chosen, the app is hardcoded to update to the latest successful workflow run from the `dev` branch. It would be nice if users got shown a drop down list and could select which branch to update to. This would be achieved by using the GitHub API.

## GitHub API

[https://api.github.com/repos/stackotter/delta-client/branches](https://api.github.com/repos/stackotter/delta-client/branches) returns a list of all branches on the repo.

The rest of this issue just requires a basic knowledge of SwiftUI to show a dropdown picker in a way that makes it clear that it is for choosing which branch the unstable update will use.

### PR DESCRIPTION
**Pull request by Matthew15625**
# Description

Fixes #8
- Added picker to select unstable branch

There's a slight problem with it. When you select a branch from the picker it greys out as if it is disabled, even though it isn't. It looks like this is just a SwiftUI bug but I will keep my eye out for any solutions.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation comments
- [x] My changes generate no new warnings

### CODE REVIEW

File: Sources/Client/Util/Updater.swift [hunk 1]
```diff
@@ -159,10 +162,10 @@ public class Updater: ObservableObject {
   // MARK: Helper
   
   /// - Returns: A download URL and a version string
-  public static func getDownloadURL(_ type: UpdateType) throws -> (URL, String) {
+  public func getDownloadURL(_ type: UpdateType) throws -> (URL, String) {
     switch type {
       case .stable:
-        return try getLatestStableDownloadURL()
+      return try Self.getLatestStableDownloadURL()
```
Comment (stackotter) [inline Sources/Client/Util/Updater.swift:168]: indent this line one level

File: Sources/Client/Util/Updater.swift [hunk 2]
```diff
@@ -234,6 +230,44 @@ public class Updater: ObservableObject {
...
+        }
+        
+        self.branches = branches
+      } catch {
+        DeltaClientApp.modalError("Failed to get list of branches from workflow runs; \(error)", safeState: .serverList)
+      }
+    }
+  }
+  
+  /// Geta a list of all workflow runs
```
Comment (stackotter) [inline Sources/Client/Util/Updater.swift:253]: typo

File: Sources/Client/Views/Config/BranchButton.swift [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import SwiftUI
+
+struct BranchButton: View {
+
```
Comment (stackotter) [inline Sources/Client/Views/Config/BranchButton.swift:4]: Remove this new line

**Review by stackotter — CHANGES_REQUESTED** [review]
Just a few formatting things to change. But the code quality is really good! Nice work 👍

**Review by stackotter — APPROVED** [review]

**Review by stackotter — APPROVED** [review]
Thanks for improving the error handling too :) that's always appreciated

**Comment by Matthew15625** [comment]
Wasn't working for some reason so I've tried again at #65 (I'm sure it works this time)

**Pull request closed without merging**